keycloak/sms-provider/src/main/java/org/sunbird/keycloak/rest/RequiredActionLinkProvider.java

```diff
@@ -0,0 +1,212 @@
+   */
+  @POST
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response generateRequiredActionLink(Map<String, String> request) {
+    logger.debug("RestResourceProvider:generateRequiredActionLink: called ");
+
+    checkRealmAdminAccess();
+    
+    String redirectUri = request.get(Constants.REDIRECT_URI);
+    String clientId = request.get(Constants.CLIENT_ID);
+    String actionName = request.get(Constants.REQUIRED_ACTION);
+    String userName = request.get(Constants.USERNAME);
+    
+    validateClientId(clientId);
+   
+    UserModel user = getUserByUserName(userName);
+    
+    ClientModel client = session.getContext().getRealm().getClientByClientId(clientId);
+    if (client == null || !client.isEnabled()) {
+      throw new WebApplicationException(
+          ErrorResponse.error(clientId + Constants.ERROR_NOT_ENABLED, Status.BAD_REQUEST));
+    }
+    
+    validateRedirectUri(redirectUri, client);
+
+    int expirationInSecs = getExpirationInSecs(request.get(Constants.EXPIRATION_IN_SECS));
+    int expiration = Time.currentTime() + expirationInSecs;
+    
+    List<String> requiredActionList = getRequiredActionList(actionName);
+
+    try {
+      ExecuteActionsActionToken token = new ExecuteActionsActionToken(user.getId(), expiration,
+          requiredActionList, redirectUri, clientId);
+      UriBuilder builder = LoginActionsService.actionTokenProcessor(session.getContext().getUri());
+      builder.queryParam(Constants.KEY,
+          token.serialize(session, session.getContext().getRealm(), session.getContext().getUri()));
+
+      String link = builder.build(session.getContext().getRealm().getName()).toString();
+      
+      Map<String, Object> response = new HashMap<>();
+      response.put(Constants.LINK, link);
+      return Response.ok(response).build();
+    } catch (Exception e) {
+      return ErrorResponse.error(Constants.ERROR_CREATE_LINK, Status.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  private UserModel getUserByUserName(String userName) {
+    logger.debug("RestResourceProvider: getUserByUserName called");
+    if (StringUtils.isBlank(userName)) {
+      throw new WebApplicationException(ErrorResponse.error(
+          MessageFormat.format(Constants.ERROR_MANDATORY_PARAM_MISSING, userName, Constants.USERNAME),
+          Status.BAD_REQUEST));
+    }
+    UserModel user = KeycloakModelUtils.findUserByNameOrEmail(session,
+        session.getContext().getRealm(), userName);
+    
+    if (user == null) {
+      throw new WebApplicationException(ErrorResponse.error(
+          MessageFormat.format(Constants.ERROR_INVALID_PARAMETER_VALUE, userName, Constants.USERNAME),
+          Status.BAD_REQUEST));
+    }
+
+    if (!user.isEnabled()) {
+      throw new WebApplicationException(
+          ErrorResponse.error(Constants.ERROR_USER_IS_DISABLED, Status.BAD_REQUEST));
+    }
+    
+    return user;
+  }
+
+  private List<String> getRequiredActionList(String actionName) {
```

Comment:
Suggest similar rename as above indicating that calling this method may cause the application to return an error.